zk-sdk/src/range_proof/mod.rs

```diff
@@ -1,3 +1,6 @@
+// Copyright (c) 2018 Chain, Inc.
+// This code is licensed under the MIT license.
```

Comment:
Could you add a link to the source and the full license if it's all copied?

Might be worth getting legal approval on this to make sure we're doing the right thing, but that can be done in follow-up work.

If you've modified it though, then it becomes a derivative work, and I think that's credited differently

Reply:
I do link the dalek-cryptography repository 5 lines below (which should also contain the license) and also mention the main modifications:
- the dalek implementation has support for MPC, which we don't need so our code is simplified slightly
- we suppose non-interactive range proof batching, which the dalek implementation doesn't have

I'll double check with legal.